src/main.rs

```diff
@@ -1,274 +1,216 @@
 
-    Some(Counter::new(l, ml, based, cringe, message.author.name))
-}
+        'outer: for page in 1.. {
+            eprintln!("Fetching page {} (after {})", page, time_utils::snowflake_time(first_id));
+            let msgs = channel_id
+                .messages(&ctx.http, |retriever| retriever.after(first_id).limit(100))
+                .await
+                .unwrap();
+            eprintln!("Retrieved {} messages", msgs.len());
 
-fn generate_thread_name(name: String, t: &str) -> String {
-    return if name.is_empty() {
-        let isoweek = &Utc::now().naive_utc().iso_week();
-        format!("{} WN {} {}", t, isoweek.week(), isoweek.year())
-    } else {
-        name
-    }
-}
+            first_id = match msgs.first() {
+                Some(first) => first.id,
+                None => break,
+            };
 
-async fn generate_thread_messages(ctx: &Context, counts: Vec<Counter>, name: String) {
-    let thread = create_thread(ctx, name).await;
-
-    println!("Start populating thread.");
-
-    for (idx, item) in counts.iter().enumerate() {
-        if idx > 14 {break;}
-        thread.send_message(&ctx.http, |msg| {
-            msg.content(format!("{}: {}", idx+1, &item.author))
-        }).await.expect("Unable to send message.");
-
-        thread.send_message(&ctx.http, |msg| {
-            msg.content(format!("{}", &item.link))
-        }).await.expect("Unable to send message.");
-
-        thread.send_message(&ctx.http, |msg| {
-            msg.add_embed(|embed| {
-                embed.title("  ");
-                embed.description(
-                    format!("{} <:based:748564944449962017> {} <:cringe:748564944819060856>    [link]({})",
-                            &item.based, &item.cringe, &item.msg_link)
-                )
-            })
-        }).await.expect("Unable to send message.");
-    }
+            for msg in &msgs {
+                if msg.timestamp > end_time {
+                    break 'outer;
+                }
+                toplist.append(msg);
+            }
+        }
 
-    println!("Done populating thread.");
-}
+        eprintln!("Finished collecting messages");
+        toplist
+    }
 
-async fn create_thread(ctx: &Context, name: String) -> GuildChannel {
-    let channel_id = ChannelId(CHANNEL);
-    let channel = channel_id.to_channel(&ctx.http)
-        .await
-        .unwrap()
-        .guild()
-        .unwrap();
-    channel.create_private_thread(&ctx.http, |thread| {
-        thread.name(name);
-        thread.auto_archive_duration(10080);
-        thread.kind(ChannelType::PublicThread)
-    }).await.expect("No permissions to create thread!")
-}
+    async fn post_toplist_thread(&self, ctx: &Context, emoji: &Option<Emoji>, list: &BTreeSet<MsgWrap>) -> Result<(), SerenityError> {
+        eprintln!("Creating thread for {:?}", emoji);
+        let thread = self.create_thread(ctx, emoji).await;
+
+        eprintln!("Starting to populate thread for {:?}", emoji);
```

Comment:
Is it intended that you use `eprintln` for non error messages?

Reply:
Yes, I use `eprintln` for messages that are only intended  to log the process and do not actually produce output.